src/libfetchers/tests/attrs.cc

```diff
@@ -0,0 +1,284 @@
+#include "attrs.hh"
+#include <nlohmann/json.hpp>
+#include <gtest/gtest.h>
+
+using json = nlohmann::json;
+
+namespace nix::fetchers {
+
+    /* ----------------------------------------------------------------------------
+     *jsonToAttrs 
+     * --------------------------------------------------------------------------*/
+
+    TEST(jsonToAttrs, simpleJson) {
+        json j = {
+            {"num", 42 },
+            {"string", "this is a string" },
+            {"bool", true }
+        };
+
+        Attrs attrs = jsonToAttrs(j);
+        auto it = attrs.find("num");
+        ASSERT_NE(it, attrs.end());
+        ASSERT_EQ(std::get<int64_t>(it->second), 42);
+
+        it = attrs.find("string");
+        ASSERT_NE(it, attrs.end());
+        ASSERT_EQ(std::get<std::string>(it->second), "this is a string");
+
+        it = attrs.find("bool");
+        ASSERT_NE(it, attrs.end());
+        //ASSERT_EQ(std::get<Explicit<bool>>(it->second).t, true); // unexpected index exception
```

Comment:
I could use some help with this. I really don't understand what's going on here and why i am failing to get the bool value out of there